src/import/v8cpuFormatter.ts

```diff
@@ -0,0 +1,61 @@
+import {CPUProfile, CPUProfileNode} from './chrome'
+
+export interface OldCPUProfileNode {
+  functionName: string
+  lineNumber: number
+  scriptId: string
+  url: string
+  hitCount: number
+  bailoutReason: string
+  id: number
+  children: OldCPUProfileNode[]
+}
+
+export interface OldCPUProfile {
+  startTime: number
+  endTime: number
+  head: OldCPUProfileNode
+  samples: number[]
+  timestamps: number[]
+}
+
+function treeToArray(node: OldCPUProfileNode, nodes: CPUProfileNode[]) {
+  nodes.push({
+    id: node.id,
+    callFrame: {
+      columnNumber: 0,
+      functionName: node.functionName,
+      lineNumber: node.lineNumber,
+      scriptId: node.scriptId,
+      url: node.url,
+    },
+    hitCount: node.hitCount,
+    children: node.children.map(child => child.id),
+  })
+  node.children.forEach(child => {
+    return treeToArray(child, nodes)
+  })
+  return nodes
+}
+
+function timestampsToDelta(timestamps: number[], startTime: number): number[] {
+  return timestamps.reduce((deltas: number[], timestamp: number, index: number) => {
```

Comment:
It doesn't look to me like you need to use `reduce` here? Could this be more concisely expressed like so?

```
return timestamps.map((timestamp, index) => {
  const lastTimestamp = index === 0 ? startTime * 1000000 : timestamps[index - 1]
  return timestamp - lastTimestamp
})
```